src/AppBundle/Entity/Post.php

```diff
@@ -203,17 +203,16 @@ public function getComments()
 
     public function addComment(Comment $comment)
     {
+        $comment->setPost($this);
         if (!$this->comments->contains($comment)) {
             $this->comments->add($comment);
-            $comment->setPost($this);
         }
     }
 
     public function removeComment(Comment $comment)
     {
-        if ($this->comments->removeElement($comment)) {
```

Comment:
Yes, you're correct. But here's another example to show that this code a bit unexpected behavior:
```php
$comment = new Comment();
$post->addComment($comment); // Comment::$post === $post object

$post->getComments()->removeElement($comment);
$post->removeComment($comment); // Comment::$post is still equal to $post object here due to the failed check in this method
```